lib/benefits_claims/providers/benefits_claims/lighthouse_benefits_claims_provider.rb

```diff
@@ -0,0 +1,169 @@
+# frozen_string_literal: true
+
+require 'benefits_claims/providers/benefits_claims/benefits_claims_provider'
+require 'benefits_claims/responses/claim_response'
+require 'lighthouse/benefits_claims/service'
+
+module BenefitsClaims
+  module Providers
+    # Provider implementation for Lighthouse Benefits Claims API
+    #
+    # Wraps the existing BenefitsClaims::Service and transforms Lighthouse API
+    # responses through the ClaimResponse DTO to ensure data consistency and validation.
+    #
+    # While Lighthouse already returns data in JSON:API format with camelCase attributes,
+    # this transformation layer demonstrates the provider pattern for future implementations
+    # (e.g., CHAMPVA) that will need to transform their native formats.
+    #
+    # @example Usage
+    #   provider = LighthouseBenefitsClaimsProvider.new(user)
+    #   claims = provider.get_claims # Returns transformed claim data
+    #   claim = provider.get_claim('123') # Returns transformed single claim
+    class LighthouseBenefitsClaimsProvider
+      include BenefitsClaimsProvider
+
+      def initialize(user)
+        @user = user
+        @service = BenefitsClaims::Service.new(user.icn)
+      end
+
+      def get_claims
+        response = @service.get_claims
+
+        # Transform each claim through the DTO
+        response['data'] = response['data'].map { |claim_data| transform_to_dto(claim_data) }
+
+        response
+      rescue Common::Exceptions::BaseError => e
```

Comment:
we have a service execption class for lighthouse specifically that will give much better errors and bubble up to the frontend cleanly.

lib/lighthouse/service_exception.rb

see this bit of code here on how to use it 
https://github.com/department-of-veterans-affairs/vets-api/blob/f58536fd7b0ba5042d90402d68d9dd6b403b7e4f/lib/lighthouse/benefits_claims/service.rb#L333